cmd/reconciler/start/service/reconciler.go

```diff
@@ -11,7 +13,14 @@ func StartComponentReconciler(ctx context.Context, o *reconCli.Options, reconcil
 	if o.DryRun {
 		service.EnableReconcilerDryRun()
 	}
-	recon, err := reconCli.NewComponentReconciler(o, reconcilerName)
+
+	durationMetric := metrics.NewComponentProcessingDurationMetric(o.Logger())
+	err := prometheus.Register(durationMetric.Collector)
+	if err != nil {
+		return nil, nil, err
+	}
+	reconcilerMetricsSet := metrics.NewReconcilerMetricsSet(durationMetric)
+	recon, err := reconCli.NewComponentReconciler(o, reconcilerName, reconcilerMetricsSet)
```

Comment:
As we are always enabling this metric and no config-values are needed, can't we directly create the MetricSet inside of the `ComponentReconciler` struct (or even in the `NewInstall` fct)?

Reply:
The MetricSet only need when componentReconciler started in web service mode, for other usage, it's not necessary to have. like in `pkg/reconciler/instances/busolamigrator/busola.go`, it's not for web service mode, right? Or am I understood wrong...
```
func init() {
	log := logger.NewLogger(false)

	log.Debugf("Initializing component reconciler '%s'", ReconcilerName)
	reconciler, err := service.NewComponentReconciler(ReconcilerName)
	if err != nil {
		log.Fatalf("Could not create '%s' component reconciler: %s", ReconcilerName, err)
	}
```